jscomp/bsb_exe/rescript_main.ml

```diff
@@ -26,7 +26,7 @@ let () = Bsb_log.setup ()
 
 let separator = "--"
 
-let watch_mode = ref false
```

Comment:
I've separated the watch mode from other delegated commands to handle it separately. So now we have better control over it, and the ref is unnecessary.

Before:
- Trigger build and exit with 1
- Start watching changes
- Trigger build

After:
- Trigger build
- Start watching changes